packages/client/helpers/functional-test/run-tests.ts

```diff
@@ -225,13 +227,18 @@ async function main(): Promise<number | void> {
 
   try {
     if (args['--updateSnapshot']) {
+      const providers = args['--provider'] ?? [...allProviders]
       const snapshotUpdate = jestCli.withArgs(['-u']).withArgs(args['_'])
-      snapshotUpdate.withEnv({ UPDATE_SNAPSHOTS: 'inline' }).run()
+
+      for (const provider of providers) {
+        snapshotUpdate.withEnv({ UPDATE_SNAPSHOTS: 'inline', ONLY_TEST_PROVIDERS: provider }).run()
```

Comment:
```suggestion
        # runs the snapshot update for each provider separately
        # because jest does not allow updating many inline snapshots in the same test in one go
        snapshotUpdate.withEnv({ UPDATE_SNAPSHOTS: 'inline', ONLY_TEST_PROVIDERS: provider }).run()
```